sdk/inc/azure/iot/az_iot_provisioning_client.h

```diff
@@ -423,6 +423,45 @@ AZ_NODISCARD az_result az_iot_provisioning_client_query_status_get_publish_topic
     size_t mqtt_topic_size,
     size_t* out_mqtt_topic_length);
 
+/**
+ * @brief Azure IoT Provisioning Client options for az_iot_provisioning_client_get_request_payload.
+ *        Not currently used.  Reserved for future use.
+ *
+ */
+typedef void az_iot_provisioning_client_payload_options;
+
+/**
+ * @brief Builds the optional payload for a provisioning request.
+ * @remark Use this API to build an MQTT payload during registration.
+ *         This call is optional for most scenarios.  Some service
+ *         applications may require #custom_payload_property specified during
```

Comment:
The `#` here won't link to the parameter. You can use backticks to give it a typewriter font. Links can be generated to functions, enums, typedefs, etc. Parameters can't be resolved since they aren't necessarily unique across all headers.
```suggestion
 *         applications may require `custom_payload_property` specified during
```